sandbox/win/src/module_file_name_interception.cc

```diff
@@ -0,0 +1,126 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/sandbox/win/src/module_file_name_interception.h"
+
+#include <string.h>
+#include <algorithm>
+#include <string>
+#include <type_traits>
+
+#include "base/logging.h"
+#include "base/strings/string_piece.h"
+#include "base/strings/string_piece_forward.h"
+#include "base/strings/string_util.h"
+#include "base/win/windows_types.h"
+#include "sandbox/win/src/sandbox_types.h"
+
+namespace {
+
+template <typename CharT>
+struct BraveToChrome;
+
+template <>
+struct BraveToChrome<char> {
+  static constexpr const base::StringPiece kBrave = "brave.exe";
+  static constexpr const base::StringPiece kChrome = "chrome.exe";
+
+  static void ReplaceAt(char* dest, size_t dest_size) {
+    ::strncpy_s(dest, dest_size, kChrome.data(), kChrome.length());
+  }
+};
+
+template <>
+struct BraveToChrome<wchar_t> {
+  static constexpr const base::WStringPiece kBrave = L"brave.exe";
+  static constexpr const base::WStringPiece kChrome = L"chrome.exe";
+
+  static void ReplaceAt(wchar_t* dest, size_t dest_size) {
+    ::wcsncpy_s(dest, dest_size, kChrome.data(), kChrome.length());
+  }
+};
+
+template <typename StringType,
+          typename CharT = std::remove_pointer_t<StringType>>
+DWORD PatchFilename(StringType filename, DWORD length, DWORD size) {
+  if (!base::EndsWith(filename, BraveToChrome<CharT>::kBrave,
+                      base::CompareCase::INSENSITIVE_ASCII)) {
+    return length;
+  }
+
+  constexpr DWORD kBraveLen = BraveToChrome<CharT>::kBrave.length();
+  constexpr DWORD kChromeLen = BraveToChrome<CharT>::kChrome.length();
+  static_assert(kBraveLen <= kChromeLen);
+  constexpr DWORD kLenDiff = kChromeLen - kBraveLen;
+
+  const size_t brave_pos = length - kBraveLen;
+  BraveToChrome<CharT>::ReplaceAt(filename + brave_pos, size - brave_pos);
+  if (size < length + kLenDiff) {
```

Comment:
i guess it's better to do this check first, then replace? Also you can return after `setlasterror`?

Reply:
No, you're wrong, read msdn

Reply:
write a comment then